packages/drivers/iframe-driver/src/innerDocumentService.ts

```diff
@@ -22,16 +25,32 @@ export class InnerDocumentService implements IDocumentService {
     /**
      * Create a new InnerDocumentService
      */
-    public static async create(proxyObject: ICombinedDriver): Promise<InnerDocumentService> {
-        return new InnerDocumentService(proxyObject, proxyObject.clientId);
+    public static async create(
+        proxyObject: ICombinedDriver,
+        logger?: ITelemetryBaseLogger,
+    ): Promise<InnerDocumentService> {
+        return new InnerDocumentService(proxyObject, proxyObject.clientId, logger);
     }
 
+    private readonly logger: ITelemetryBaseLogger;
+
     private constructor(
         private readonly outerProxy: ICombinedDriver,
-        public clientId: string) { }
+        public clientId: string,
+        logger?: ITelemetryBaseLogger)
+    {
+        // outerProxy provides a logger as well, but we can create and use our own here
+        // for different functionality
+        this.logger = logger
+            ? ChildLogger.create(logger, "InnerIFrameDriver")
+            : new IFrameDebugLogger("InnerIFrameDriver");
+    }
 
     // TODO: Issue-2109 Implement detach container api or put appropriate comment.
     public get resolvedUrl(): IResolvedUrl {
+        const event = { category: "generic", eventName: "not implemented" };
+        this.outerProxy.logger.send(event);
+        this.logger.send(event);
```

Comment:
Is this intentionally sending twice?  I think I'm also not really understanding in what situations this should be logging on its own logger vs. trying to access the outerProxy's logger.  It feels a little strange to me to be directly accessing another object's logger.

Reply:
it's sending twice so both its own and the outer proxy logger get used.  it's definitely not "right" but more for the example atm.

i don't think it's _that_ crazy when the other object is the object we're proxying.  it seems like we might want to log from the outer driver when the inner driver doesn't need its own logger (like in our first party case) and log from the inner when it does (custom 3rd part logger)?  the main thing was that both options should be available if either can ever a better call than the other, but if we think one is just always going to be better then removing the option seems fine